pandas/tests/indexes/conftest.py

```diff
@@ -47,3 +49,40 @@ def zero(request):
     # For testing division by (or of) zero for Index with length 5, this
     # gives several scalar-zeros and length-5 vector-zeros
     return request.param
+
+
+def _get_subclasses(cls):
+    for subclass in cls.__subclasses__():
+        yield from _get_subclasses(subclass)
+        yield subclass
+
+
+all_indexes_inc_abc = [Index] + list(set(_get_subclasses(Index)))
+all_indexes_inc_abc_sorted = sorted(all_indexes_inc_abc,
+                                    key=attrgetter('__name__'))
+all_indexes = [index for index in all_indexes_inc_abc_sorted
+               if getattr(pd, index.__name__, False)]
+
+
+@pytest.fixture(params=all_indexes)
+def all_index_types(request):
+    """
+    A Fixture for all indexes types. Index and subclasses in pandas namespace.
+    """
+    return request.param
+
+
+@pytest.fixture
```

Comment:
shouldn't this just work off of the indices_list? why creating extra machinery here?